src/components/ComponentPreview.stories.tsx

```diff
@@ -143,14 +146,15 @@ export const Email = {
   },
 
   play: async ({canvasElement, args}) => {
-    const canvas = await within(canvasElement);
+    const canvas = within(canvasElement);
 
     // check that the user-controlled content is visible
     await canvas.findByText('Email preview');
     await canvas.findByText('A preview of the email Formio component');
 
     // check that the input name is set correctly
-    const input = await canvas.getByLabelText('Email preview');
+    const input = canvas.getByLabelText('Email preview');
+    // @ts-ignore
```

Comment:
`component.key` - it does not properly propagate that `args.component` satisfies the AnyComponentSchema and thinks it may be `FallbackSchema` . I tried `expect(someTypePredicate(args.component)).toBe(true)` but that also doesn't help with type narrowing, because TS doesn't realize execution stops on failed exception.